inc/Task.php

```diff
@@ -120,65 +108,89 @@ private function saveFigures($figure) {
                     msg('invalid or empty extenson figure: ' . $type, -1);
                     continue;
                 }
-                $name = $this->getImagePath($type);
+                $name = $this->getAttachmentPath($figure['caption'], $type);
                 if (io_saveFile(mediaFN($name), (string)trim($imgContent))) {
-                    msg('image: ' . $name . ' for language ' . $this->lang . ' has been saved', 1);
+                    msg('Figure "' . $figure['caption'] . '" for language ' . $this->lang . ' has been saved', 1);
+                } else {
+                    msg('Figure "' . $figure['caption'] . '" for language ' . $this->lang . ' has not saved properly!', -1);
                 }
                 $this->figures[] = ['path' => $name, 'caption' => $figure['caption']];
             }
         }
     }
 
     /**
-     * @param string $type
-     * @return string
+     * Returns ID path of the Attachment based on its caption
+     * @param $caption string Attachment Caption
+     * @param $type string File type
+     * @return string ID
      */
-    private function getImagePath($type = null) {
-        if ($type) {
-            return $this->getPluginName() . ':figure:year' . $this->year . '_series' . $this->series . '_' . $this->label . '_' .
-                $this->lang . '.' . $type;
-        }
-        return $this->getPluginName() . ':figure:year' . $this->year . '_series' . $this->series . '_' . $this->label . '_' . $this->lang;
+    private function getAttachmentPath($caption, $type) {
+        $name = substr(preg_replace("/[^a-zA-Z0-9_-]+/", '-', $caption), 0, 30) . '_' . substr(md5($caption.$type),0,5);
+        return vsprintf($this->helper->getConf('attachment_path_' . $this->lang), [$this->year, $this->series, $this->label]) . ':' . $name . '.' . $type;
     }
 
-    public function getFileName() {
-        $id = 'fkstaskrepo:' . $this->year . ':' . $this->series . '-' . $this->label . '_' . $this->lang;
-        return metaFN($id, '.dat');
+    /**
+     * Returns the path of .json file with task data.
+     * @return string path of file
+     */
+    private function getFileName() {
+        global $conf;
+        return $conf['metadir'] . '/' . vsprintf($this->helper->getConf('task_data_meta_path'), [$this->year, $this->series, $this->label]);
```

Comment:
Treba použiť metaFN()

Reply:
No, ve skutečnosti ne, ale bude to lepší předělat.